### ISSUE TITLE
TypeError: unhashable type: 'list'

### ISSUE THREAD

**Issue by renefritze** [tarmack/Pythagora#2]
running 026a352a31e47aba2f07e83ab1441ec694e25987 on python 2.6.5 I get:

./pythagora 
error:  [Errno 2] No such file or directory: '/home/kosh/Music/shoutcast-bookmarks.xml'
Traceback (most recent call last):
  File "/share/projekte/pythagora/LibraryForm.py", line 94, in reload
    self.albumdict[album] = self.albumdict.get(album, [])+[song]
TypeError: unhashable type: 'list'

Current playlist is populated correctly, but the artist/track/album panels remain empty.
python-mpd is at 0.2.1

**Comment by tarmack** [comment]
Apparantly the album tag in one of your songs contains multiple values. This is allowed by the spec but was not handled. It should be solved with commit a8b468d. This is not a clean solution but should get you going for now.

**Comment by tarmack** [comment]
I had another look at the code of my fix, it will not help you a bit I'm afraid. I think I was already sleeping.
With commit 460d344 it should actualy work. Please let me know if you still have any problems with it.

**Comment by renefritze** [comment]
Thanks for looking into this. I did get a little further, but ended up with another TypeError in line 96 this time. I did write a couple lines to get around that: 6a159d54e3591a05e0a720b18d3523105bc8008f, but I'm not too sure that's getting the intention of the code right. Panels look good now though. f317d861f9a5b048d59db943ece257fa7cd24fe9 fixes my usage of tabs instead of  spaces. 
If you need more looking into this/testing lemme know, I'd be happy to help.

**Comment by tarmack** [comment]
Indeed my code is not correct, if artist has a list value the interpreter tries to create a hash of it for checking against the dictionary keys, but it still can' t hash a list.
I worked on it again. It now converts all keys and values to lists before processing so I'm pretty confident this time I actually got it right.
Some insight in the reason for existence of the albumlist, which is probably not named very descriptive, sorry:
The intent of the albumlist dictionary is to collect a list of all artists involved. Thus the list should not contain nested lists. currently these artist lists are only used to display a tooltip over the album listing all artists involved.
